charts/auto-scaling-runner-set/templates/autoscalingrunnerset.yaml

```diff
@@ -12,6 +12,23 @@ spec:
   runnerGroup: {{ . }}
   {{- end }}
 
+  {{- if .Values.proxy }}
+  proxy:
+    {{- if .Values.proxy.http }}
+    http:
+      url: {{ .Values.proxy.http.url }}
+      credentialSecretRef: {{ .Values.proxy.http.credentialSecretRef }}
+    {{ end }}
+    {{- if .Values.proxy.https }}
+    https:
+      url: {{ .Values.proxy.https.url }}
+      credentialSecretRef: {{ .Values.proxy.https.credentialSecretRef }}
+    {{ end }}
+    {{- if and .Values.proxy.noProxy (kindIs "slice" .Values.proxy.noProxy) }}
+    noProxy: {{ .Values.proxy.noProxy | toYaml | nindent 6}}
+    {{ end }}
+  {{ end }}
+
```

Comment:
This gets rendered to 

```yaml
spec:
  # ...
  proxy:
    http:
      url: http://proxy:3128
      credentialSecretRef: proxy-auth

    https:
      url: http://proxy:3128
      credentialSecretRef: proxy-auth

    noProxy:
      - example.com
      - example.org
```

Reply:
do you mind adding an L0 test for this in `charts/auto-scaling-runner-set/tests/template_test.go`?

Reply:
do you also need to update the `values.yaml` to provide some example input behind comments?

Reply:
Will do as soon as we're set on the spec 👌